### PR TITLE
Make a move

### PR DESCRIPTION
**Pull request by LiveNathan**
Just black stones on a black board for now.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,48 +1,75 @@
...
-    const boardSize = 10;
-    let halfSize = boardSize / 2;
-    let boardPlaneGeometry = new THREE.PlaneGeometry(boardSize, boardSize);
+window.addEventListener('click', onClick, false);
 
-    // Add the game board as a simple plane first.
-    let boardPlane = new THREE.Mesh(boardPlaneGeometry, material);
-    boardPlane.rotation.x = Math.PI / 2;
-    boardPlane.position.set(halfSize, -0.01, halfSize);
+function onClick(event) {
```
Comment (iansedano) [inline main.js]: Typically these functions are called "handlers", i.e. `clickHandler`. Normally, we try to reserve `onClick` for the event name.

File: main.js [hunk 2]
```diff
@@ -1,48 +1,75 @@
 import * as THREE from 'three';
```
Comment (iansedano) [inline main.js:1]: ooohhh Three.js 👀

Never used it myself, but definitely heard a lot about it.

Also, looks like you've got imports and npm working, great job!

File: main.js [hunk 3]
```diff
@@ -1,48 +1,75 @@
...
+window.addEventListener('click', onClick, false);
 
-    // Add the game board as a simple plane first.
-    let boardPlane = new THREE.Mesh(boardPlaneGeometry, material);
-    boardPlane.rotation.x = Math.PI / 2;
-    boardPlane.position.set(halfSize, -0.01, halfSize);
+function onClick(event) {
+    // Normalized mouse coordinates
+    mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
+    mouse.y = -(event.clientY / window.innerHeight) * 2 + 1;
```
Comment (iansedano) [inline main.js:38]: Having to handle all the x and y positions could make this project much more complex that what was intended, but we're on that track now, let's see where it leads :)

**Comment by iansedano** [comment]
Something to note about the HTML (which hasn't been changed in this PR), is that since you're using `type="module"` in the script tag, this means that the script is automatically deferred. That is, it's left until after the HTML has finished loading. This means you can put it in the `head`, which is typically neater than having to put it at the end of the `body`.

See the `defer` entry in here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script

**Review by iansedano — APPROVED** [review]
Looks very cool!

image

The stones don't seem to get placed in the right location, which could be a mini-challenge to get working right.

I did something like this with canvas and my solution was to define a grid of boxes with their center on the intersection of lines. If the click was within one of the boxes, then that's where the stone was placed.

Very interested to see how this develops! I do worry that we'll get a bit sidetracked into getting three.js to play nice, but it's cool, so let's run with it :)

File: main.js (hunk 2 above)
Comment (LiveNathan) [reply main.js:1]: Yes! I'm hoping to use some 3d design in SubAligner, so this seemed like a good opportunity to try it.

**Comment by LiveNathan** [comment]
> Looks very cool!
> 
> image
> 
> The stones don't seem to get placed in the right location, which could be a mini-challenge to get working right.
> 
> I did something like this with canvas and my solution was to define a grid of boxes with their center on the intersection of lines. If the click was within one of the boxes, then that's where the stone was placed.
> 
> Very interested to see how this develops! I do worry that we'll get a bit sidetracked into getting three.js to play nice, but it's cool, so let's run with it :)

Yeah, I'm trying not to get into the weeds with the Three.js styling and stuff. Just trying to focus mostly on the mechanics of the game.

**Pull request merged**